packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -440,10 +442,9 @@ async function setupOrgBrowser(
 
 export async function activate(context: vscode.ExtensionContext) {
   const extensionHRStart = process.hrtime();
-  // Telemetry
   const machineId =
     vscode && vscode.env ? vscode.env.machineId : 'someValue.machineId';
-  telemetryService.initializeService(context, machineId);
+  await telemetryService.initializeService(context, machineId);
```

Comment:
I think this is still a costly function to await on in `activate`.
1. One approach I could think of is make `isTelemetryEnabled` of `TelemetryService` an async function, and do something like:
```
  public sendExtensionActivationEvent(hrstart: [number, number]): void {
    this.isTelemetryEnabled().then(()=>{
        // ...
    });
```
This requires a bigger refactor though.

2. Another approach is to move this down further. My main concern is about the time for commands to become available (`registerCommands`).

cc @lcampos

Reply:
@xyc I've measured this and refactoring it is not a high priority. The highest impact item in the activation process are other checks which we are addressing as part of the perf work. Once we're done with the perf work a lot of the checks here will go away.